app/formaters/DesFormatter.scala

```diff
@@ -49,7 +49,7 @@ class DesFormatter @Inject() (addressFormater: AddressFormter, requestSupport: R
 
     val transactionIterable = for {
       fdAllOption <- financialData
-    } yield (fdAllOption.financialTransactions.filter(f => (f.periodKey == periodKey.value && f.chargeType == "VAT Return Debit Charge")))
+    } yield (fdAllOption.financialTransactions.filter(f => (f.periodKey.getOrElse("ZZ") == periodKey.value && f.chargeType == "VAT Return Debit Charge")))
```

Comment:
There is nice `sealed trait` with `case object`s which models the `ChargeType`. It would be type safe using it instead of inline strings. Have a look:

https://github.com/hmrc/direct-debit-backend/blob/6a1c46fbed1e43bd5bcd6a5b0bdca2bc194b644a/app/model/alert/PushNotificationRequest.scala#L69